### PR TITLE
Support setting redirect uri.

### PR DESCRIPTION
**Pull request by han-gao**
New features:  Support user to set redirect-uri.

### CODE REVIEW

File: e -i HEAD~2 [hunk 1]
```diff
@@ -0,0 +1,254 @@
+[33mcommit ec132e144a6f60accba0750aae473a63db3c58ea[m[33m ([m[1;36mHEAD -> [m[1;32mEnable_property_azure.activedirectory.redirect-uri-template[m[33m)[m
```
Comment (stliu) [inline e -i HEAD~2:1]: why you changed this file

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java [hunk 2]
```diff
@@ -427,7 +420,8 @@ public void afterPropertiesSet() throws Exception {
                             .filter(type -> !AADAuthorizationGrantType.AUTHORIZATION_CODE.equals(type))
                             .findAny()
                             .ifPresent(notUsed -> {
-                                throw new IllegalStateException("onDemand only support authorization_code grant type. ");
```
Comment (stliu) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java:430]: this change is not necessary

File: sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java [hunk 3]
```diff
@@ -54,18 +54,9 @@
     private String userNameAttribute;
 
     /**
-     * @deprecated Now the redirect-url-template is not configurable.
```
Comment (stliu) [inline sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java:57]: so, this is a regression?

**Comment by stliu** [comment]
could you link this PR to the github issue?

**Comment by chenrujun** [comment]
Hi, @han-gao , please resolve the conflicts.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 4]
```diff
@@ -570,6 +570,50 @@ In [Resource server visiting other resource server] scenario(For better descript
         return "Response from WebApiB.";
     }
     ```  
+
+#### Support setting redirect-uri-template.
+
+This starter supports Reset Redirection Endpoint(authorization server to return responses containing authorization credentials to the client via the resource owner user-agent). Developers can customize the redirect-uri.
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:576]: Delete `This starter supports Reset Redirection Endpoint(authorization server to return responses containing authorization credentials to the client via the resource owner user-agent).`,

Just keep `Developers can customize the redirect-uri.`

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 5]
```diff
@@ -570,6 +570,50 @@ In [Resource server visiting other resource server] scenario(For better descript
...
+
+![what-is-redirect-uri](resource/what-is-redirect-uri.png)
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:578]: Delete the text in the picture. Just point out the `redirect_uri`.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 6]
```diff
@@ -570,6 +570,50 @@ In [Resource server visiting other resource server] scenario(For better descript
...
+* Step 1: Add `redirect-uri-template` properties in application.yml.
+    ```yaml
+        azure:
+          activedirectory:
+            redirect-uri-template: --your-redirect-uri-template--
+    ```
+
+* Step 2: Update the configuration of the azure cloud platform in the portal.
+
+We need to configure the same redirect-uri as application.yml:
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:589]: Keep 4 spaces at the beginning of this line.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 7]
```diff
@@ -570,6 +570,50 @@ In [Resource server visiting other resource server] scenario(For better descript
...
+
+![web-application-config-redirect-uri](resource/web-application-config-redirect-uri.png)
+
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:592]: Same here.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 8]
```diff
@@ -570,6 +570,50 @@ In [Resource server visiting other resource server] scenario(For better descript
...
+
+* Step 3: Write your Java code:
+
+After we set redirect-uri-template, we need to update `SecurityConfigurerAdapter`:
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:596]: Here.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 9]
```diff
@@ -570,6 +570,50 @@ In [Resource server visiting other resource server] scenario(For better descript
...
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class AADOAuth2LoginSecurityConfig extends AADWebSecurityConfigurerAdapter {
+    /**
+     * Add configuration logic as needed.
+     */
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        super.configure(http);
+        http.oauth2Login()
+                .loginProcessingUrl("/{your-path-prefix}")
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:609]: Why `your-path-prefix`? Maybe it should be `your redirect uri`?

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**